ui/apps/platform/src/Containers/ComplianceEnhanced/Dashboard/ScanResultsToolbar.tsx

```diff
@@ -0,0 +1,180 @@
+import React, { useState } from 'react';
+import {
+    Pagination,
+    SearchInput,
+    SelectOption,
+    Toolbar,
+    ToolbarContent,
+    ToolbarItem,
+} from '@patternfly/react-core';
+
+import SelectSingle from 'Components/SelectSingle';
+import { UsePaginationResult } from 'hooks/patternfly/usePagination';
+
+export type SearchFilter = {
+    scanName: string;
+    clusterName: string;
+    profileName: string;
+};
+
+type FilterInputProps = {
+    setSearchFilter: React.Dispatch<React.SetStateAction<SearchFilter>>;
+};
+
+type ProfileFilterInputProps = FilterInputProps & {
+    profileFilter: string;
+    setProfileFilter: React.Dispatch<React.SetStateAction<string>>;
+};
+
+type ClusterFilterInputProps = FilterInputProps & {
+    clusterFilter: string;
+    setClusterFilter: React.Dispatch<React.SetStateAction<string>>;
+};
+
+function ProfileFilterInput({
+    profileFilter,
+    setSearchFilter,
+    setProfileFilter,
+}: ProfileFilterInputProps) {
+    return (
+        <ToolbarItem variant="search-filter">
+            <SearchInput
+                aria-label="Filter by profile name"
+                placeholder="Filter by profile name"
+                value={profileFilter}
+                onChange={(_e, selection: string) => setProfileFilter(selection)}
+                onSearch={() =>
+                    setSearchFilter((prev) => ({ ...prev, profileName: profileFilter }))
+                }
+                onClear={() => {
+                    setProfileFilter('');
+                    setSearchFilter((prev) => ({ ...prev, profileName: '' }));
+                }}
+            />
+        </ToolbarItem>
+    );
+}
+
+function ClusterFilterInput({
+    clusterFilter,
+    setSearchFilter,
+    setClusterFilter,
+}: ClusterFilterInputProps) {
+    return (
+        <ToolbarItem variant="search-filter">
+            <SearchInput
+                aria-label="Filter by cluster name"
+                placeholder="Filter by cluster name"
+                value={clusterFilter}
+                onChange={(_e, selection: string) => setClusterFilter(selection)}
+                onSearch={() =>
+                    setSearchFilter((prev) => ({ ...prev, clusterName: clusterFilter }))
+                }
+                onClear={() => {
+                    setClusterFilter('');
+                    setSearchFilter((prev) => ({ ...prev, clusterName: '' }));
+                }}
+            />
+        </ToolbarItem>
+    );
+}
+
+type ScanResultsToolbarProps = {
+    numberOfScanResults: number;
+    searchFilter: SearchFilter;
+    setSearchFilter: React.Dispatch<React.SetStateAction<SearchFilter>>;
+} & UsePaginationResult;
```

Comment:
Good call defining this type in terms of the other type 👍 

Nit: I'd also recommend exporting the component's prop type too. If we end up passing props a few levels deep, having the exported prop makes it easier to avoid duplication and refactoring if something changes.

Reply:
I'd actually argue against this change. I agree we want to avoid duplication but when I see export in front of something, my assumption is that it is definitely being used somewhere else. The other downside is if you remove the reference to an exported definition you loss out on the `@typescript-eslint/no-unused-vars` linter error.

Reply:
I usually feel the same way, but I've flipped my stance for React component props specifically.

- The calling component frequently needs to know the types of the arguments it is passing in, or even declare them locally
- Exporting the type doesn't increase the bundle size at all, since the types are erased during build
- It feels very unlikely that we will delete the component itself, but leave the prop types behind. Usually if the component goes the whole file goes with it.

Either way though, if it turned out we need the prop type exported, it is a simple one line change if/when that becomes the case.